ldclient_scoped.go

```diff
@@ -0,0 +1,612 @@
+	key string,
+	defaultVal ldvalue.Value,
+) (ldvalue.Value, error) {
+	return c.client.JSONVariationCtx(ctx, key, c.CurrentContext(), defaultVal)
+}
+
+// JSONVariationDetail is the same as [LDScopedClient.JSONVariation], but also
+// returns further information about how the value was calculated. The "reason"
+// data will also be included in analytics events.
+//
+// For more information, see the Reference Guide:
+// https://docs.launchdarkly.com/sdk/features/evaluation-reasons#go
+func (c *LDScopedClient) JSONVariationDetail(key string, defaultVal ldvalue.Value) (ldvalue.Value, ldreason.EvaluationDetail, error) {
+	return c.client.JSONVariationDetail(key, c.CurrentContext(), defaultVal)
+}
+
+// JSONVariationDetailCtx is the same as [LDScopedClient.JSONVariationCtx], but
+// also returns further information about how the value was calculated. The
+// "reason" data will also be included in analytics events.
+//
+// Cancelling the context.Context will not cause the evaluation to be cancelled.
+// The context.Context is used by hook implementations refer to [ldhooks.Hook].
+//
+// For more information, see the Reference Guide:
+// https://docs.launchdarkly.com/sdk/features/evaluation-reasons#go
+func (c *LDScopedClient) JSONVariationDetailCtx(
+	ctx gocontext.Context,
+	key string,
+	defaultVal ldvalue.Value,
+) (ldvalue.Value, ldreason.EvaluationDetail, error) {
+	return c.client.JSONVariationDetailCtx(ctx, key, c.CurrentContext(), defaultVal)
+}
+
+// MigrationVariation returns the migration stage of the migration feature flag
+// for the current evaluation context.
+//
+// Returns defaultStage if there is an error or if the flag doesn't exist.
+//
+// For more information, see the Reference Guide: https://docs.launchdarkly.com/sdk/features/evaluating#go
+func (c *LDScopedClient) MigrationVariation(
+	key string,
+	defaultStage ldmigration.Stage,
+) (ldmigration.Stage, interfaces.LDMigrationOpTracker, error) {
+	return c.client.MigrationVariation(key, c.CurrentContext(), defaultStage)
+}
+
+// MigrationVariationCtx is the same as [LDScopedClient.MigrationVariation], but
+// accepts a context.Context.
+//
+// Cancelling the context.Context will not cause the evaluation to be cancelled.
+// The context.Context is used by hook implementations refer to [ldhooks.Hook].
+//
+// For more information, see the Reference Guide:
+// https://docs.launchdarkly.com/sdk/features/evaluating#go
+func (c *LDScopedClient) MigrationVariationCtx(
+	ctx gocontext.Context,
+	key string,
+	defaultStage ldmigration.Stage,
+) (ldmigration.Stage, interfaces.LDMigrationOpTracker, error) {
+	return c.client.MigrationVariationCtx(ctx, key, c.CurrentContext(), defaultStage)
+}
+
+// Identify sends an identify event for the current evaluation context.
+//
+// For more information, see the Reference Guide:
+// https://docs.launchdarkly.com/sdk/features/identify#go
+func (c *LDScopedClient) Identify() error {
+	return c.client.Identify(c.CurrentContext())
+}
```

Comment:
I'm honestly not sure if `Identify` needs a "scoped" variant.

In general, scoped methods are not necessary because you can always call `scopedClient.Client().SomeMethod(..., scopedClient.CurrentContext(), ...)`. I didn't copy over `AllFlagsState` or `SecureModeHash`, despite those being methods that currently take a context.

Reply:
I agree. Having this method here I think would only add confusion. I think it's better to remove it entirely.